### DISCUSSION TITLE
Jinja or django template engine?

### DISCUSSION THREAD

**Discussion by Weissnix4711** [inventree/InvenTree#3652] [Q&A] [answered]
Hi,

Which templating engine is used for labels? The docs [would suggest Jinja](https://inventree.readthedocs.io/en/stable/report/labels/#custom-labels). However, after I tried using recursive for loops, a feature which django does not to my knowledge have, but jinja does, I looked at the produced error in the admin section: `django.template.exceptions.TemplateSyntaxError`. That would suggest to me that django is being used, not jinja?

Could someone clarify what is going on here?

Thanks,
Thomas Aldrian

**Comment by matmair** [comment, marked as answer]
@Weissnix4711 Not sure why it says jinja. Weasyprint is using Django's templating engine - I will fix the docs.

**Reply by Weissnix4711** [reply]
Thanks
